src/core/mesh/qgsmesh3daveraging.h

```diff
@@ -140,6 +140,7 @@ class CORE_EXPORT QgsMesh3dAveragingMethod SIP_ABSTRACT
     virtual void volumeRangeForFace(
       double &startVerticalLevel,
       double &endVerticalLevel,
+      int &singleVerticalLevel,
```

Comment:
Is change of CORE api required with addition of startVerticalLevel? (applies for all functions in this PR) 

I suggest 2 alternatives 
1) keep the OLD API and let caller use startVerticalLevel == endVerticalLevel for this case
2) create new functions without startVerticalLevel & endVerticalLevel ? since they are not used in this situation.

Reply:
Maybe I am missing something, but:
For 1., we don't have always `startVerticalLevel == endVerticalLevel` when there is only one volume. With that case, `startVerticalLevel` is the level of the top of the volume, and `endVerticalLevel` the level of the bottom of the volume.

For 2., where `volumeRangeForFace()` is called, in the abstract class level, we will don't know which function to call because we know nothing about the type of averaging method.

Then  `volumeRangeForFace()` is private, so not sure it is considered as a part of the CORE API.